sdk/turing/router/config/router_ensembler_config.py

```diff
@@ -76,11 +77,12 @@ def docker_config(self, docker_config: turing.generated.models.EnsemblerDockerCo
         if isinstance(docker_config, turing.generated.models.EnsemblerDockerConfig):
             self._docker_config = docker_config
         elif isinstance(docker_config, dict):
-            docker_config['resource_request'] = \
-                turing.generated.models.ResourceRequest(**docker_config['resource_request'])
-            docker_config['env'] = [turing.generated.models.EnvVar(**env_var) for env_var in docker_config['env']]
+            openapi_docker_config = docker_config.copy()
+            openapi_docker_config['resource_request'] = \
+                turing.generated.models.ResourceRequest(**openapi_docker_config['resource_request'])
+            openapi_docker_config['env'] = [turing.generated.models.EnvVar(**env_var) for env_var in openapi_docker_config['env']]
```

Comment:
I've made some additional changes - they look pretty minor but they were causing some bugs in the unit tests. 

Essentially, the previous implementation of the setter methods were mutating the dictionaries that were being passed to it, which is dangerous and might lead to unintended effects. These new changes simply create new working dictionaries that are only used within the scope of the setter.

cc @krithika369

Reply:
Sounds good. Thanks for addressing them. We need to propagate these changes to main later as well.